qt/scientific_interfaces/Muon/MuonAnalysis.cpp

```diff
@@ -2035,6 +2038,44 @@ void MuonAnalysis::selectMultiPeak(const QString &wsName) {
   selectMultiPeak(wsName, boost::optional<QString>());
 }
 
+/**
+* Enable PP tool for the plot of the given WS. Does not update data selector
+* @param wsName Name of the WS which plot PP tool will be attached to.
+* is for "load current run" where the data file has a temporary name like
+* MUSRauto_E.tmp
+*/
+void MuonAnalysis::selectMultiPeakNoUpdate(const QString &wsName) {
+  disableAllTools();
+  if (!plotExists(wsName)) {
+    plotSpectrum(wsName);
+    setCurrentDataName(wsName);
+  }
+
+  if (wsName != m_fitDataPresenter->getAssignedFirstRun()) {
+    // Set the available groups/pairs and periods
+    const Grouping groups = m_groupingHelper.parseGroupingTable();
+    QStringList groupsAndPairs;
+    groupsAndPairs.reserve(
+        static_cast<int>(groups.groupNames.size() + groups.pairNames.size()));
+    std::transform(groups.groupNames.begin(), groups.groupNames.end(),
+                   std::back_inserter(groupsAndPairs), &QString::fromStdString);
```

Comment:
It looks like this conversion to `QString` is done twice, once here and once in `setGroupsAndPairs`?

Reply:
it does look like. I cannot recall why I did it that way